src/NUnitEngine/Addins/nunit.v2.driver.tests/XmlExtensionsTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// ***********************************************************************
+// Copyright (c) 2016 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using NUnit.Core;
+using NUnit.Framework;
+
+namespace NUnit.Engine.Drivers.Tests
+{
+    [TestFixture]
+    public class XmlExtensionsTests
+    {
+        private TestResult _result;
+
+        [SetUp]
+        public void BuildResult()
+        {
+            _result = new TestResult(TestName.Parse("Fake Test"));
+            _result.SetResult(ResultState.Failure, "]]>", string.Empty);
+        }
+
+        [Test]
+        public void CDataIsEscaped()
+        {
+            string innerXml = _result.ToXml(false).InnerXml;
+            string expected = "<message><![CDATA[]]]]><![CDATA[>]]></message>";
+            Assert.That(innerXml, Is.StringContaining(expected));
+        }
+    }
+}
```

Comment:
The existing v2 driver test is an integration test. Specifically, it's a sampling of some of NUnit V2's own tests, referencing the V2 framework and being run using the driver. It's isolated in it's own directory so that there is no possibility of it's referencing anything except the V2 framework.

This test, which you have added, is obviously a unit test. We should undoubtedly have such tests but at the time I was creating the driver, I decided to focus on the integration/system level of testing. The problem is that by adding this single test, you had to also add a reference from the tests to the v2 driver, which invalidates the overall integration test. If we want to create unit tests for this driver, they have to be in some other assembly, perhaps in the overall addins test assembly. Alternatively, we can wait until we actually separate the build of the driver from NUnit itself, which I suspect we will eventually do.